### PR TITLE
[JBPM-8998] Adding test classes for "Missing SpringBoot support in NotificationListener"

### PR DESCRIPTION
**Pull request by afalhambra**
Related PRs: 

- https://github.com/kiegroup/droolsjbpm-integration/pull/1992/files
- https://github.com/kiegroup/jbpm/pull/1587/files

### CODE REVIEW

**Comment by afalhambra** [comment]
> Looks good to me @afalhambra
> Just an enhancement: to speed up the test, use a different process with a lower deadline notification (2-3 seconds, for ex)

Agree. Will create a separate BPM process with a deadline notification of up to 3secs.

File: kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/listeners/CountDownLatchNotificationListener.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.jbpm.services.task.deadlines.NotificationListener;
+import org.kie.internal.task.api.UserInfo;
+import org.kie.internal.task.api.model.NotificationEvent;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CountDownLatchNotificationListener implements NotificationListener {
+
+    private CountDownLatch latch;
+    private static List<NotificationEvent> eventsReceived = new ArrayList<NotificationEvent>();
```
Comment (gmunozfe) [inline kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/listeners/CountDownLatchNotificationListener.java]: This shouldn't be static, and it's initialized at constructor (I think that's the reason of test failure at CI);
```suggestion
    private List<NotificationEvent> eventsReceived;
```

Comment (afalhambra) [reply kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/listeners/CountDownLatchNotificationListener.java]: Done, waiting for Jenkins to rebuild

**Comment by elguardian** [comment]
failure is not related but let jenkis try again.
Jenkins retest this please.

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
@gmunozfe plz review.

**Comment by elguardian** [comment]
@kiegroup/gatekeepers plz merge

**Pull request merged**